ibmmq/mqiMQCNO.go

```diff
@@ -26,6 +26,29 @@ package ibmmq
 #include <cmqc.h>
 #include <cmqxc.h>
 
+void freeCCDTUrl(MQCNO *mqcno) {
+#if defined(MQCNO_VERSION_6) && MQCNO_CURRENT_VERSION >= MQCNO_VERSION_6
+	if (mqcno->CCDTUrlPtr != NULL) {
+		free(mqcno->CCDTUrlPtr);
+	}
+#endif
+}
+
+void setCCDTUrl(MQCNO *mqcno, PMQCHAR url, MQLONG length) {
+#if defined(MQCNO_VERSION_6) && MQCNO_CURRENT_VERSION >= MQCNO_VERSION_6
+	mqcno->CCDTUrlOffset = 0;
+	mqcno->CCDTUrlPtr = NULL;
+	mqcno->CCDTUrlLength = length;
+	if (url != NULL) {
+		mqcno->CCDTUrlPtr = url;
```

Comment:
Need to forcibly set the cno version here so the user doesn't have to do it ...

if (mqcno->Version < MQCNO_VERSION_6) {
   mqcno->Version = MQCNO_VERSION_6;
}

Reply:
Thanks!